src/libexec/zmqstat

```diff
@@ -133,6 +133,22 @@ sub processQ {
                                 } else {
                                         $qf{TIME} = $data;
                                 }
+                        } elsif ($rec eq "p") {
+                                my $jump;
```

Comment:
Given get_record comes from qshape (ref: https://github.com/vdukhovni/postfix/blob/master/postfix/auxiliary/qshape/qshape.pl), I feel like the pointer handling should be the same as https://github.com/vdukhovni/postfix/blob/master/postfix/auxiliary/qshape/qshape.pl#L263.  If that were the case, it seems like if $data were > 0 we could just call seek and be done with things.  But we are checking $ix >= -1 and then using substr instead, and I can't determine why were doing something different.  Perhaps an example would help.  With out that, maybe we can get a description of what were addressing and how this differs from qshape.

Reply:
@plobbes : something like this then, right?
```suggestion
                        } elsif ($rec eq "p") {
                                my $jump = $data; 
                                if($jump > 0) {
                                        seek($fh,$jump,0);
                                } else {
                                        next;
                                }
```

Reply:
Unless I'm missing something, I think we can use $data directly (no need for $jump).
Looks like we should return/handle error unless seek() is successful.
Also, I don't believe we need an else/next here.

Reply:
You are certainly correct about $jump, it was there merely to make the purpose of the contents of $data more readable to anyone perusing the code. I've mimicked more closely the behaviour of the code of qshape.pl . I've also tried it out with a queue file which I've truncated just to see what would happen with a failing seek. The code just moves to the next file, which I think is desirable when parsing an unknown amount of files. Tell me if you see anything else in need of corrections/improvement.